extensions-core/hdfs-storage/src/main/java/io/druid/storage/hdfs/HdfsDataSegmentKiller.java

```diff
@@ -45,7 +45,7 @@
   public HdfsDataSegmentKiller(final Configuration config, final HdfsDataSegmentPusherConfig pusherConfig)
   {
     this.config = config;
-    this.storageDirectory = new Path(pusherConfig.getStorageDirectory());
+    this.storageDirectory = new Path(HdfsDataSegmentUtil.getFullyQualifiedHdfsPath(pusherConfig.getStorageDirectory(), config));
```

Comment:
kill works by taking the location from loadSpec, which would already have the namenode uri in there or else it will use the default namenode by itself.